From a1240fa791d39c1247eafa8f1aabcf3eb71127db Mon Sep 17 00:00:00 2001
From: Mateusz Belicki <mateusz.belicki@intel.com>
Date: Fri, 22 Oct 2021 09:09:42 +0000
Subject: [PATCH]  Fixed incorrect cast in JointMatrixResolution.

Incorrect cast caused the other branch of if-else expression to be
effictively dead, instead of handling the case when given type is not a
vector. This commit addresses issue #209.
---
 .../Optimizer/OpenCLPasses/JointMatrixFuncsResolutionPass.cpp   | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/IGC/Compiler/Optimizer/OpenCLPasses/JointMatrixFuncsResolutionPass.cpp b/IGC/Compiler/Optimizer/OpenCLPasses/JointMatrixFuncsResolutionPass.cpp
index 1d77e6870..17d27818f 100644
--- a/IGC/Compiler/Optimizer/OpenCLPasses/JointMatrixFuncsResolutionPass.cpp
+++ b/IGC/Compiler/Optimizer/OpenCLPasses/JointMatrixFuncsResolutionPass.cpp
@@ -204,7 +204,7 @@ static Type *getIntegerEquivalent(Type *matTy) {
         return matTy;
     }
 
-    if (IGCLLVM::FixedVectorType *VT = cast<IGCLLVM::FixedVectorType>(matTy)) {
+    if (IGCLLVM::FixedVectorType *VT = dyn_cast<IGCLLVM::FixedVectorType>(matTy)) {
         unsigned elements = (unsigned) VT->getNumElements();
         unsigned size = VT->getElementType()->getScalarSizeInBits();
         Type *elementType = Type::getIntNTy(matTy->getContext(), size);
-- 
2.32.0

